src/Sulu/Component/Webspace/Manager/WebspaceCollectionBuilder.php

```diff
@@ -352,7 +352,7 @@ private function buildUrlPartialMatch(
                 $portal->getWebspace()->getDefaultSegment(),
                 $urlRedirect,
                 $urlAnalyticsKey,
-                $url->isMain(),
+                false, // partial matches cannot be main
```

Comment:
partial matches shouldn't be main portal-informations because they should not be used to generate urls (to avoid unnecessary 301 redirects)